upstairs/src/lib.rs

```diff
@@ -648,9 +629,32 @@ async fn proc(
 
                     }
                     Some(Message::YouAreNoLongerActive(new_active_uuid)) => {
-                        if up.uuid != new_active_uuid {
-                            up.set_inactive();
+                        /*
+                         * XXX If we get this response, but the new uuid is
+                         * actually our uuid, do we just ignore it?
+                         */
+                        println!(
+                            "[{}] {} downstairs self deactivate: new {:?}",
+                            up.uuid,
+                            new_active_uuid,
+                            up_coms.client_id
+                        );
+                        up.ds_transition(up_coms.client_id, DsState::New);
+                        up.set_inactive();
+                        if up.uuid == new_active_uuid {
+                            /*
+                             * Now, this is really going off the rails.  Our
+                             * downstairs things we have a different UUID
+                             * and is sending us the UUID of the "new"
+                             * downstairs, but that UUID IS our UUID.  So
+                             * clearly the downstairs is confused.
+                             */
+                            bail!(
+                                "[{}] {} bad YouAreNoLongerActive, {:?}!",
+                                up_coms.client_id, up.uuid, new_active_uuid,
+                            );
                         }
+                        return Err(CrucibleError::UuidMismatch.into());
```

Comment:
To check my understanding, returning this Err + setting this client_id to DsState::New will cause a whole new connection attempt to that particular downstairs, which means that this upstairs will attempt to negotiate and go active again?

Reply:
Yes, it will cause a whole new connection for this downstairs.
Because we `set_inactive()` just above, this upstairs will now require a new activate request from the guest before the downstairs can re-join.

Reply:
:+1: